core/commonMain/src/implementations/immutableList/SmallPersistentVector.kt

```diff
@@ -36,6 +36,7 @@ internal class SmallPersistentVector<E>(private val buffer: Array<Any?>) : Immut
     }
 
     override fun addAll(elements: Collection<E>): PersistentList<E> {
+        if (elements.isEmpty()) return this
```

Comment:
What do you think about adding the same check to other persistent collections? 
Even though they eventually return the same instance, they create a throwaway builder in the case of an empty argument.